frontend/src/_actions/user.actions.js

```diff
@@ -0,0 +1,97 @@
+import { userConstants } from '../_constants';
+import { userService } from '../_services';
+import { alertActions } from './';
+import { history } from '../_helpers';
+
+export const userActions = {
+    login,
+    logout,
+    register,
+    getAll,
+    delete: _delete
+};
+
+function login(username, password) {
+    return dispatch => {
+        dispatch(request({ username }));
+
+        userService.login(username, password)
+            .then(
```

Comment:
Thank you for using Promise properly.
I’m not sure if you're not getting bad http code. Would it be success really?